plugins/inputs/snmp/snmp.go

```diff
@@ -623,6 +623,11 @@ func (s *Snmp) getConnection(idx int) (snmpConnection, error) {
 	gs := gosnmpWrapper{&gosnmp.GoSNMP{}}
 	s.connectionCache[idx] = gs
 
+	if strings.HasPrefix(agent, "tcp:") {
+		parts := strings.SplitN(agent, ":", 2)
+		gs.Transport = parts[0]
+		agent = parts[1]
```

Comment:
Not checking the length here made me nervous, but I guess at the worst, parts[1] would only be blank.

Reply:
It is safe, because the HasPrefix check, but I'd probably use strings.TrimPrefix to remove it instead.